src/helpers/fetch.ts

```diff
@@ -51,60 +51,78 @@ export async function fetchGraphql<TResult>(
   return result.data as Promise<TResult>;
 }
 
-export function fetchGetWithinPage<TResult>(page: Page, url: string): Promise<TResult | null> {
-  return page.evaluate(innerUrl => {
-    return new Promise<TResult | null>((resolve, reject) => {
-      fetch(innerUrl, {
-        credentials: 'include',
-      })
-        .then(result => {
-          if (result.status === 204) {
-            resolve(null);
-          } else {
-            resolve(result.json());
-          }
-        })
-        .catch(e => {
-          reject(e);
-        });
-    });
+export async function fetchGetWithinPage<TResult>(
+  page: Page,
+  url: string,
+  ignoreErrors = false,
+): Promise<TResult | null> {
+  const [result, status] = await page.evaluate(async innerUrl => {
+    let response: Response | undefined;
+    try {
+      response = await fetch(innerUrl, { credentials: 'include' });
+      if (response.status === 204) {
+        return [null, response.status] as const;
+      }
+      return [await response.text(), response.status] as const;
+    } catch (e) {
+      throw new Error(
+        `fetchGetWithinPage error: ${e instanceof Error ? `${e.message}\n${e.stack}` : String(e)}, url: ${innerUrl}, status: ${response?.status}`,
+      );
+    }
   }, url);
+  if (result !== null) {
+    try {
+      return JSON.parse(result);
+    } catch (e) {
+      if (!ignoreErrors) {
+        throw new Error(
+          `fetchGetWithinPage parse error: ${e instanceof Error ? `${e.message}\n${e.stack}` : String(e)}, url: ${url}, result: ${result}, status: ${status}`,
+        );
+      }
+    }
+  }
+  return null;
 }
```

Comment:
Wow the code became complicated, can you add a description to the PR with a use case example?

Reply:
Done